src-control/Control/RefCount.hs

```diff
@@ -476,42 +483,51 @@ finaliserRefTracker inner refid allocSite = do
           Just (refid', _) | refid < refid' -> return ()
           _ -> writeIORef globalForgottenRef (Just (refid, allocSite))
 
-assertNoForgottenRefs :: IO ()
-assertNoForgottenRefs = do
+getForgottenRef :: IO (Maybe RefException)
```

Comment:
It might be simpler to have this just be
```
assertNoForgottenRefs :: MonadThrow m => m ()
```
and use `unsafeIOToPrim` internally _just_ for the `IORef` operations, and nothing else.

Reply:
Done